### PR TITLE
Docstring fixes for `entrypoints` and `experiment_trackers` modules

### PR DESCRIPTION
**Pull request by strickvl**
I made the relevant fixes for the `entrypoints` and `experiment_trackers` modules, bringing the docstrings in line with our new style.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/entrypoints/step_entrypoint_configuration.py [hunk 1]
```diff
@@ -228,17 +242,22 @@ def get_entrypoint_command(cls) -> List[str]:
...
-        """Gets all the options that are required when running an entrypoint
-        with this configuration.
+        """Gets all the options that are required when running an entrypoint with this configuration.
 
         **Note**: Subclasses should implement the
             `get_custom_entrypoint_options()` class method instead of this
             one if they require custom options.
+
+        Returns:
+            A set of strings with all the options that are required when running.
```
Comment (fa9r) [inline src/zenml/entrypoints/step_entrypoint_configuration.py:260]: > 80 chars

File: src/zenml/entrypoints/step_entrypoint_configuration.py [hunk 2]
```diff
@@ -228,17 +242,22 @@ def get_entrypoint_command(cls) -> List[str]:
...
+        Returns:
+            A list of strings with the command.
         """
         return DEFAULT_SINGLE_STEP_CONTAINER_ENTRYPOINT_COMMAND
 
     @classmethod
     def get_entrypoint_options(cls) -> Set[str]:
-        """Gets all the options that are required when running an entrypoint
-        with this configuration.
+        """Gets all the options that are required when running an entrypoint with this configuration.
```
Comment (fa9r) [inline src/zenml/entrypoints/step_entrypoint_configuration.py:253]: >80 chars

File: src/zenml/entrypoints/step_entrypoint_configuration.py [hunk 3]
```diff
@@ -177,11 +186,16 @@ def get_run_name(self, pipeline_name: str) -> str:
...
+            pipeline_name: The name of the pipeline.
+
+        Returns:
+            The run name.
         """
 
     def setup(self, pipeline_name: str, step_name: str) -> None:
-        """Runs setup code that needs to run before any user code is imported
-        or the step is executed.
+        """Runs setup code that needs to run before any user code is imported or the step is executed.
```
Comment (fa9r) [inline src/zenml/entrypoints/step_entrypoint_configuration.py:198]: >80 chars

File: src/zenml/entrypoints/step_entrypoint_configuration.py [hunk 4]
```diff
@@ -132,14 +134,16 @@ def __init__(self, arguments: List[str]):
 
     @classmethod
     def get_custom_entrypoint_options(cls) -> Set[str]:
-        """Custom options for this entrypoint configuration that are
-        required in addition to the default ZenML ones.
+        """Custom options for this entrypoint configuration that are required in addition to the default ZenML ones.
```
Comment (fa9r) [inline src/zenml/entrypoints/step_entrypoint_configuration.py:137]: >80 chars

File: src/zenml/entrypoints/step_entrypoint_configuration.py [hunk 5]
```diff
@@ -11,6 +11,8 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 #  or implied. See the License for the specific language governing
 #  permissions and limitations under the License.
+"""Abstract base class for entrypoint configurations that run a single step."""
```
Comment (fa9r) [inline src/zenml/entrypoints/step_entrypoint_configuration.py:14]: >80 chars

**Review by fa9r — CHANGES_REQUESTED** [review]
Several lines are >80 chars now and there is one change in _parse_arguments that seems odd. Otherwise looks good!

File: src/zenml/entrypoints/step_entrypoint_configuration.py [hunk 6]
```diff
@@ -401,9 +424,9 @@ def _parse_arguments(cls, arguments: List[str]) -> Dict[str, Any]:
         Returns:
             Dictionary of the parsed arguments.
 
-        Raises:
+        Raise:
             ValueError: If the parsing failed because the argument format is
-                incorrect or one of the options is missing.
+                        incorrect or one of the options is missing.
```
[the remark this answers is not in the document]
Comment (strickvl) [reply src/zenml/entrypoints/step_entrypoint_configuration.py:429]: Some issue with `darglint`. I've fixed it by adding in an ignore comment for this error on this specific function.

File: src/zenml/entrypoints/step_entrypoint_configuration.py (hunk 5 above)
Comment (strickvl) [reply src/zenml/entrypoints/step_entrypoint_configuration.py:14]: 73 characters.

**Review by htahir1 — APPROVED** [review]

**Pull request merged**